test/src/Main.idr

```diff
@@ -10,32 +10,38 @@ import System.FFI
 import System.File
 import System.File.Buffer
 
-abs : (i, q : Int8) -> Int16
+rescaleU8 : Int8 -> Double
+rescaleU8 x = (cast {to = Double} (x - 128)) / 128.0
+
+abs : (i, q : Int8) -> Double
 abs i q =
   let
     ii : Double
-    ii = cast i * cast i
+    ii = rescaleU8 i * rescaleU8 i
 
     qq : Double
-    qq = cast q * cast q
+    qq = rescaleU8 q * rescaleU8 q
   in
-    cast $ sqrt ( ii + qq )
+    cast $ sqrt ( (ii + qq) / 2 )
```

Comment:
this should be `sqrt ( ii + qq ) * 0.5 * sqrt(2.0)` so that the rescale happens in Double type.